### PR TITLE
Add values_file to helm-upgrade-from-source

### PR DESCRIPTION
**Pull request by NissesSenap**
# Changes

Making it easier to overwrite multiple values,
Instead of defining them all in overwrite_values.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

**Review by vinamra28 — CHANGES_REQUESTED** [review]

File: task/helm-upgrade-from-source/0.2/helm-upgrade-from-source.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      description: "The values file to be used"
+      default: "values.yaml"
+    - name: helm_version
+      description: "Specify a specific helm version"
+      default: "latest"
+  workspaces:
+    - name: source
+  steps:
+    - name: upgrade
+      image: docker.io/lachlanevenson/k8s-helm:$(inputs.params.helm_version)
```
[the remark this answers is not in the document]
Comment (vinamra28) [reply task/helm-upgrade-from-source/0.2/helm-upgrade-from-source.yaml:40]: can you please change this in script as well :slightly_smiling_face:

[the remark this answers is not in the document]
Comment (NissesSenap) [reply task/helm-upgrade-from-source/0.2/helm-upgrade-from-source.yaml:40]: Ahh of course, no problem.
Remind me in which version did we go to v1beta1?
Do we need to update:
tekton.dev/pipelines.minVersion: "0.12.1"
as well?

File: task/helm-upgrade-from-source/0.2/samples/helm-upgrade-from-repo-task-run.yaml.tmpl [hunk 2]
```diff
@@ -0,0 +1,18 @@
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: example-helm-upgrade-from-repo
+spec:
+  taskRef:
```
[the remark this answers is not in the document]
Comment (NissesSenap) [reply task/helm-upgrade-from-source/0.2/samples/helm-upgrade-from-repo-task-run.yaml.tmpl]: Agreed

File: task/helm-upgrade-from-source/0.2/helm-upgrade-from-source.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (vinamra28) [reply task/helm-upgrade-from-source/0.2/helm-upgrade-from-source.yaml]: nope...that's okay

[the remark this answers is not in the document]
Comment (NissesSenap) [reply task/helm-upgrade-from-source/0.2/helm-upgrade-from-source.yaml]: Fixed it.

I also changed to a sha256 base image as you have started to use it in the rest of the tasks.
I also broke out docker.io/lachlanevenson/k8s-helm, and made it into a variable container_repo.

Is it better to have it all in a single parameter or should we split it up like I have done now?

**Comment by NissesSenap** [comment]
I also want to point out:  --values "$(params.charts_dir)/$(params.values_file)"
I assume that the values file always will be in charts_dir.

Doing it this way is definitely more user friendly but it's a assumption that might create issue for certain users.

**Review by vinamra28 — APPROVED** [review]
can you please squash the commits :slightly_smiling_face:

File: task/helm-upgrade-from-source/0.2/helm-upgrade-from-source.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (vinamra28) [reply task/helm-upgrade-from-source/0.2/helm-upgrade-from-source.yaml]: ahh...yes please a single param for image would be better

[the remark this answers is not in the document]
Comment (NissesSenap) [reply task/helm-upgrade-from-source/0.2/helm-upgrade-from-source.yaml]: Updated it now, used helm_version to be as similar as possible to 0.1
But the variable name might be a bit misleading.

File: task/helm-upgrade-from-source/0.2/README.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+#### Parameters
+
+- **charts_dir**: The directory in the source repository where the installable chart should be found.
+- **release_version**: The version of the release (*default: v1.0.0*)
+- **release_name**: The name of the release (*default: helm-release*)
+- **release_namespace**: The namespace in which the release is to be installed (*default: ""*)
+- **overwrite_values**: The values to be overwritten (*default: ""*)
+- **values_file**: The values file to be used,
+    assumes the values file is in the same dir as your charts_dir (*default: "values.yaml"*)
+- **helm_version**: The helm version which should be used (*default: latest*)
```
Comment (vinamra28) [inline task/helm-upgrade-from-source/0.2/README.md:22]: Please change the description here also. Also please do change the default value. I guess it's not latest

**Review by vinamra28 — COMMENTED** [review]
/lgtm

**Review by fhopfensperger — APPROVED** [review]
/lgtm

**Pull request merged**